### PR TITLE
update Makefile to check Plutono dashboards

### LINKED ISSUE #7761: monitoring: Make sure grafana dashboards are valid JSON files
**How to categorize this issue?**

/area monitoring
/area dev-productivity
/kind enhancement

**What would you like to be added**:
In https://github.com/gardener/gardener/issues/7747#issuecomment-1497391411 we noticed that a grafana dashboard has invalid JSON. We should add a check to `make verify` that check that these dashboard files are valid JSON documents.

**Why is this needed**:
To prevent introducing invalid grafana dashboard JSON.

### PR DESCRIPTION
**Pull request by Sallyan**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:
Sometimes Plutono dashboard has invalid JSON in syntax or not match the dashboard required fields. 
This PR updates the Makefile and adds options to verify the dashboards. Run `make check-plutono-dashboards`, it will verify all the dashboards are valid json and have the required dashboard title. 
It also fixes the dashboards which uid/title not set or duplicate UIDs. 

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/7761

**Special notes for your reviewer**:

**Release note**:

```feature developer
The plutono dashboards are now verified as part of `make check`.
```

### CODE REVIEW

**Comment by Sallyan** [comment]
cc @istvanballok

File: hack/check-plutono-dashboards.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+echo "> Checking Plutono dashboards"
+
```
Comment (shafeeqes) [inline hack/check-plutono-dashboards.sh:23]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -147,6 +147,10 @@ add-license-headers: $(GO_ADD_LICENSE)
 check-generate:
 	@hack/check-generate.sh $(REPO_ROOT)
 
+.PHONY: check-plutono-dashboards
+check-plutono-dashboards:
+	@hack/check-plutono-dashboards.sh
```
Comment (istvanballok) [inline Makefile:152]: In the original PR, this check is also performed in the `make check` call here:

https://github.com/gardener/gardener/pull/8065/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R172

Why is that change no longer part of this PR?

**Review by istvanballok — CHANGES_REQUESTED** [review]
It looks good to me besides the comment above.

File: pkg/component/plutono/dashboards/shoot/owners/kubernetes-pods-dashboard.json [hunk 3]
```diff
@@ -828,6 +828,6 @@
   },
   "timezone": "utc",
   "title": "Kubernetes Pods",
-  "uid": "kube-pods",
+  "uid": "shoot-kube-pods",
```
Comment (istvanballok) [inline pkg/component/plutono/dashboards/shoot/owners/kubernetes-pods-dashboard.json:831]: Regarding the failing unit tests: you need to execute `pkg/component/plutono/plutono_suite_test.go` and "extract" the new generated hash for the configmap name e.g. by debugging the failing test. If the dashboards are changed, the configmap suffix in the unit tests needs to be adjusted, like here:

https://github.com/gardener/gardener/pull/8390

Comment (rickardsjp) [reply pkg/component/plutono/dashboards/shoot/owners/kubernetes-pods-dashboard.json:831]: Just to note the result of a call with @Sallyan:

> To see the new hashes I added:
>```
>configmaps := &corev1.ConfigMapList{}
>c.List(ctx, configmaps)
>```
> in `pkg/component/plutono/plutono_suite_test.go` and
>```
>(*configmaps).Items[0].ObjectMeta.Name
>```
>to my watches.

which I found a more convenient way to access the hashes. The change in `pkg/component/plutono/plutono_suite_test.go` therefore needs to be as follows:

```
plutono-dashboards-garden-fbef06ba -> plutono-dashboards-garden-40b40715
plutono-dashboards-4f58da97 -> plutono-dashboards-6938ad4b
plutono-dashboards-e60246de -> plutono-dashboards-72c159d8
plutono-dashboards-bc6281a9 -> plutono-dashboards-d72a9abb
```

And for the record, I find this test problematic. I understand it doesn't specifically aim to test the names, but it's both cumbersome and going against the purpose of a test to have to copy values from the debugger in order to add them to the code.

File: pkg/component/plutono/plutono_test.go [hunk 4]
```diff
@@ -620,12 +620,12 @@ status:
 			It("should succesfully deploy all resources", func() {
 				Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-dashboard-providers-29d306e7.yaml"])).To(Equal(providerConfigMapYAMLFor(values)))
 				Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-datasources-0fd41775.yaml"])).To(Equal(dataSourceConfigMapYAMLFor(values)))
-				testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-d479d31e"}, 33)
+				testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-6938ad4b"}, 33)
```
Comment (rickardsjp) [inline pkg/component/plutono/plutono_test.go]:
```suggestion
				testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-2858c729"}, 33)
```

File: pkg/component/plutono/plutono_test.go [hunk 5]
```diff
@@ -620,12 +620,12 @@ status:
...
 				Expect(string(managedResourceSecret.Data["service__some-namespace__plutono.yaml"])).To(Equal(serviceYAMLFor(values)))
 				Expect(string(managedResourceSecret.Data["ingress__some-namespace__plutono.yaml"])).To(Equal(ingressYAMLFor(values)))
 				managedResourceDeployment, _, err := kubernetes.ShootCodec.UniversalDecoder().Decode(managedResourceSecret.Data["deployment__some-namespace__plutono.yaml"], nil, &appsv1.Deployment{})
 				Expect(err).ToNot(HaveOccurred())
-				deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-d479d31e"})
+				deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-6938ad4b"})
```
Comment (rickardsjp) [inline pkg/component/plutono/plutono_test.go]:
```suggestion
				deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-2858c729"})
```

File: pkg/component/plutono/plutono_test.go [hunk 6]
```diff
@@ -642,12 +642,12 @@ status:
 				It("should succesfully deploy all resources", func() {
 					Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-dashboard-providers-29d306e7.yaml"])).To(Equal(providerConfigMapYAMLFor(values)))
 					Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-datasources-0fd41775.yaml"])).To(Equal(dataSourceConfigMapYAMLFor(values)))
-					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-22785a17"}, 38)
+					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-72c159d8"}, 38)
```
Comment (rickardsjp) [inline pkg/component/plutono/plutono_test.go]:
```suggestion
					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-dc69f5b3"}, 38)
```

File: pkg/component/plutono/plutono_test.go [hunk 7]
```diff
@@ -642,12 +642,12 @@ status:
...
 					Expect(string(managedResourceSecret.Data["service__some-namespace__plutono.yaml"])).To(Equal(serviceYAMLFor(values)))
 					Expect(string(managedResourceSecret.Data["ingress__some-namespace__plutono.yaml"])).To(Equal(ingressYAMLFor(values)))
 					managedResourceDeployment, _, err := kubernetes.ShootCodec.UniversalDecoder().Decode(managedResourceSecret.Data["deployment__some-namespace__plutono.yaml"], nil, &appsv1.Deployment{})
 					Expect(err).ToNot(HaveOccurred())
-					deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-22785a17"})
+					deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-72c159d8"})
```
Comment (rickardsjp) [inline pkg/component/plutono/plutono_test.go]:
```suggestion
					deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-dc69f5b3"})
```

File: pkg/component/plutono/plutono_test.go [hunk 8]
```diff
@@ -661,12 +661,12 @@ status:
 				It("should succesfully deploy all resources", func() {
 					Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-dashboard-providers-29d306e7.yaml"])).To(Equal(providerConfigMapYAMLFor(values)))
 					Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-datasources-0fd41775.yaml"])).To(Equal(dataSourceConfigMapYAMLFor(values)))
-					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-aea83f97"}, 27)
+					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-d72a9abb"}, 27)
```
Comment (rickardsjp) [inline pkg/component/plutono/plutono_test.go]:
```suggestion
					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: "plutono-dashboards-d557f751"}, 27)
```

File: pkg/component/plutono/plutono_test.go [hunk 9]
```diff
@@ -661,12 +661,12 @@ status:
...
 					Expect(string(managedResourceSecret.Data["service__some-namespace__plutono.yaml"])).To(Equal(serviceYAMLFor(values)))
 					Expect(string(managedResourceSecret.Data["ingress__some-namespace__plutono.yaml"])).To(Equal(ingressYAMLFor(values)))
 					managedResourceDeployment, _, err := kubernetes.ShootCodec.UniversalDecoder().Decode(managedResourceSecret.Data["deployment__some-namespace__plutono.yaml"], nil, &appsv1.Deployment{})
 					Expect(err).ToNot(HaveOccurred())
-					deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-aea83f97"})
+					deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-d72a9abb"})
```
Comment (rickardsjp) [inline pkg/component/plutono/plutono_test.go]:
```suggestion
					deployment := deploymentYAMLFor(values, []string{"plutono-dashboards-d557f751"})
```

**Review by rickardsjp — CHANGES_REQUESTED** [review]
I think your rebase contained some changes to the dashboards. Please update the hashes in the unit test as we discussed [here](https://github.com/gardener/gardener/pull/8401#discussion_r1315577720). My criticism of this test still stands, I'll see if I we can't use `c.List` instead of `c.Get`.

**Review by rickardsjp — APPROVED** [review]
/lgtm
/retest

**Pull request merged**